common/internal_model/src/test/scala/weco/catalogue/internal_model/generators/IdentifiersGenerators.scala

```diff
@@ -1,17 +1,27 @@
 package weco.catalogue.internal_model.generators
 
+import org.scalacheck.Arbitrary
 import weco.fixtures.RandomGenerators
-import weco.catalogue.internal_model.identifiers.{
-  CanonicalId,
-  IdentifierType,
-  SourceIdentifier
-}
+import weco.catalogue.internal_model.identifiers.{CanonicalId, IdentifierType, SourceIdentifier}
 
 import scala.util.Random
 
 trait IdentifiersGenerators extends RandomGenerators {
-  def createCanonicalId: CanonicalId =
-    CanonicalId(randomAlphanumeric(length = 8).toLowerCase())
+  implicit val arbitraryCanonicalId: Arbitrary[CanonicalId] =
+    Arbitrary {
+      createCanonicalId
+    }
+
+  // We have a rule that says SourceIdentifier isn't allowed to contain whitespace,
+  // but sometimes scalacheck will happen to generate such a string, which breaks
+  // tests in CI.  This generator is meant to create SourceIdentifiers that
+  // don't contain whitespace.
```

Comment:
I get this - but it did confuse me for a second as I thought that this related to the `createCanonicalId` below. Probably just the fact they're in the same PR though.